controllers/storagecluster/external_resources.go

```diff
@@ -189,7 +189,12 @@ func newExternalGatewaySpec(rgwEndpoint string, reqLogger logr.Logger, tlsEnable
 		reqLogger.Error(err, "Host IP should not be empty in rgw endpoint")
 		return nil, err
 	}
-	gateWay.ExternalRgwEndpoints = []cephv1.EndpointAddress{{IP: hostIP}}
+	address := net.ParseIP(hostIP)
+	if address != nil {
+		gateWay.ExternalRgwEndpoints = []cephv1.EndpointAddress{{IP: hostIP}}
+	} else {
+		gateWay.ExternalRgwEndpoints = []cephv1.EndpointAddress{{Hostname: hostIP}}
+	}
```

Comment:
```suggestion
	address := net.ParseIP(hostIP)
	gateWay.ExternalRgwEndpoints = []cephv1.EndpointAddress{{Hostname: hostIP}}
	if address != nil {
		gateWay.ExternalRgwEndpoints = []cephv1.EndpointAddress{{IP: hostIP}}
	} 
```

Reply:
Updated

Reply:
Opposite way as the other one is least likely